src/core/config/mac.h

```diff
@@ -500,14 +500,14 @@
 #endif
 
 /**
- * @def OPENTHREAD_CONFIG_MAC_BEACON_PAYLOAD_PARSING_ENABLE
+ * @def OPENTHREAD_CONFIG_MAC_BEACON_PAYLOAD_ENABLE
  *
  * This setting configures if the beacon payload parsing needs to be enabled in MAC. This is optional and is disabled by
  * default because Thread 1.2.1 has removed support for beacon payloads.
  *
  */
-#ifndef OPENTHREAD_CONFIG_MAC_BEACON_PAYLOAD_PARSING_ENABLE
-#define OPENTHREAD_CONFIG_MAC_BEACON_PAYLOAD_PARSING_ENABLE 0
+#ifndef OPENTHREAD_CONFIG_MAC_BEACON_PAYLOAD_ENABLE
+#define OPENTHREAD_CONFIG_MAC_BEACON_PAYLOAD_ENABLE 0
```

Comment:
I think we should actually default this to 1.  Without this functionality new builds will not work with deployed devices.  @jwhui @abtink What do you think?

Reply:
I agree. This is a footgun if defaulted off.

Reply:
Yes, this should default to 1.

Reply:
> > > > > @gabekassel it would be great if you could try out the changes in this PR and verify that it fixes the issue you are experiencing.
> > > > 
> > > > 
> > > > @romacdon I built [this](https://github.com/openthread/ot-br-posix/compare/main...gabekassel:gkass/beacon-payload-revert?expand=1) with `-DOPENTHREAD_CONFIG_MAC_BEACON_PAYLOAD_ENABLE=1` defined in my cmake args and still see failure with the commercial device in question.
> > > 
> > > 
> > > Thanks @gabekassel. Can you provide a packet trace so that @hemanth-silabs and @suveshpratapa can debug it.
> > 
> > 
> > Attached, though not sure I'm capturing everything desired. This is wireshark using pyspinel setup with the network key. On the OTBR logging side, I just see continual beacon requests.
> > ![Screen Shot 2022-05-27 at 12 40 19 PM](https://user-images.githubusercontent.com/2759863/170779302-3855b8c4-f1c8-425e-8477-3620ed5a0031.png)
> > [beacon_payload.pcapng.zip](https://github.com/openthread/openthread/files/8789378/beacon_payload.pcapng.zip)
> 
> Thanks again @gabekassel - I'll defer to @suveshpratapa and @hemanth-silabs now to see if this is enough info to figure why the change did not solve the issue.



> > > > > @gabekassel it would be great if you could try out the changes in this PR and verify that it fixes the issue you are experiencing.
> > > > 
> > > > 
> > > > @romacdon I built [this](https://github.com/openthread/ot-br-posix/compare/main...gabekassel:gkass/beacon-payload-revert?expand=1) with `-DOPENTHREAD_CONFIG_MAC_BEACON_PAYLOAD_ENABLE=1` defined in my cmake args and still see failure with the commercial device in question.
> > > 
> > > 
> > > Thanks @gabekassel. Can you provide a packet trace so that @hemanth-silabs and @suveshpratapa can debug it.
> > 
> > 
> > Attached, though not sure I'm capturing everything desired. This is wireshark using pyspinel setup with the network key. On the OTBR logging side, I just see continual beacon requests.
> > ![Screen Shot 2022-05-27 at 12 40 19 PM](https://user-images.githubusercontent.com/2759863/170779302-3855b8c4-f1c8-425e-8477-3620ed5a0031.png)
> > [beacon_payload.pcapng.zip](https://github.com/openthread/openthread/files/8789378/beacon_payload.pcapng.zip)
> 
> Thanks again @gabekassel - I'll defer to @suveshpratapa and @hemanth-silabs now to see if this is enough info to figure why the change did not solve the issue.

Thanks @gabekassel for the logs. It looks like there is no beacon at all. Also all the packet seems to have CRC errors. Maybe the beacon requests are getting dropped due to CRC errors?

This fix comes into picture if the beacon requests are processed and beacons are sent out. Looking at the logs it looks like the beacon requests are not being processed.

If this is on OTBR is it possible for you to collect some OTBR logs? Maybe we can see what happens to beacon requests and if they are getting dropped because of `Bad FCS`

Reply:
> Yes, this should default to 1.

@romacdon @gabekassel @romacdon @edmont If we default the macro to 1, do we even need the macro? Can we get rid of it altogether?

Reply:
i do see beacons going out from the router. logs attached
[eero.log](https://github.com/openthread/openthread/files/8789580/eero.log)

Reply:
> @romacdon @gabekassel @romacdon @edmont If we default the macro to 1, do we even need the macro? Can we get rid of it altogether? Especially since with default configuration, we will see lot of issues with older devices.

Removing the configuration to transmit beacons makes sense to me. I would keep the configuration for receiving the beacon payload and default to off to discourage its use going forward.

Reply:
@gabekassel I am not sure what is going on here. The logs do show the beacons going out, but I don't see the beacons in the OTA capture. Can you let me know what commercial device this is?

Reply:
I've tested Nanoleaf Essential, Eve Energy, and Wemo Plug. All behave the same.